### DISCUSSION TITLE
algorithms/ads/11

### DISCUSSION THREAD

**Discussion by giscus** [NoughtQ/notebook#18] [Announcements]
# algorithms/ads/11

NoughtQ的笔记本

https://note.noughtq.top/algorithms/ads/11

**Comment by JazZyzJ** [comment]
对式子：

$$
    P_{opt} \leq P_{greedy} + p_{max}
$$
    
这是因为一个前提：在小数解中，其实只会有一个物品被拆分放入（因为价值大的剩余位置够用时肯定装满价值大的），也就是说贪心解和小数解的差值就是没有被拆分放入背包的物品的价值，这个物品肯定是放入背包的物品中的价值最小的一个，在这里进行放缩，将这个被拆分的物品价值替换为价值最大的物品，这时P_{greedy} + p_{max}一定大于P_{frac}，进而大于P_{opt}

**Reply by NoughtQ** [reply]
说的很有道理 :smile:，感谢补充 :pray:，之后我会更新这一部分的解释

**Comment by TillyEndless** [comment]
NFDH的效果图好像其实是FFDH的？（图上还有文字）

**Reply by NoughtQ** [reply]
草，我本来想先写NFDH的，写着写着就变成了FFDH，所以出现了凌乱的顺序，十分抱歉，我马上更新~

（甚至图片上的文字都是错的，想写FFDH，潜意识里出现了FDS🤓）
